lib/npm.js

```diff
@@ -296,16 +304,7 @@ class Npm {
       Object.assign(err, this.#getError(err, { pkg: localPkg }))
     }
 
-    // TODO: make this not need to be public
-    this.finish()
-
-    output.flush({
-      [META]: true,
-      // json can be set during a command so we send the
-      // final value of it to the display layer here
-      json: this.loaded && this.config.get('json'),
-      jsonError: jsonError(err, this),
-    })
+    this.finish(err)
```

Comment:
Did we give up on that TODO?

Reply:
I removed it because it felt weird to add more public things to a function that I had commented to make private.

I needed to add more things because of the unit tests of some utils which don't actually `exec` any commands. They do init an npm instance so the easiest thing to do was make them also call `npm.finish()` which would get called automatically after a normal `npm.exec()`.